core/unix/os.c

```diff
@@ -5019,6 +5034,9 @@ ignorable_system_call_normalized(int num)
 #    endif
     case SYS_readlinkat: return !DYNAMO_OPTION(early_inject);
 #endif
+#ifdef SYS_openat2
+    case SYS_openat2: return IS_STRING_OPTION_EMPTY(xarch_root);
```

Comment:
Hmm should the warning then be under -xarch_root too?

Reply:
Yeah, there should be. Done.

Reply:
Wait, this the syscall is ignorable if xarch_root is not set right? That is the case this PR doesn't print warning for currently. 
If xarch_root is set, the pre and post handling will print the warning anyway.

Reply:
The xarch_root path manipulation seems to be very targeted to what the loader uses: it is only looking at openat and not at open.  So if the loader switched to using open it would also find the wrong lib w/o a warning: but these are easy errors to find since you'd get the wrong arch and see the error right away; plus these are common syscalls so we don't want to intercept (and avoid inlining) if we can avoid it.

Reply:
Okay so no warning then?

Reply:
I don't have a strong opinion for the -xarch_root case, could go either way.

Reply:
Removed it.